### PR TITLE
feat: add readarr integration

### PR DESCRIPTION
**Pull request by manuel-rw**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

### CODE REVIEW

File: packages/cron-jobs/src/jobs/integrations/media-organizer.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const mediaOrganizerJob = createCronJob("mediaOrganizer", EVERY_MINUTE).w
 
       //Asserting the integration kind until all of them get implemented
       const integrationInstance = integrationCreatorFromSecrets(
-        integration as Modify<typeof integration, { kind: "sonarr" | "radarr" | "lidarr" }>,
+        integration as Modify<typeof integration, { kind: "sonarr" | "radarr" | "lidarr" | "readarr" }>,
```
Comment (Meierschlumpf) [inline packages/cron-jobs/src/jobs/integrations/media-organizer.ts:30]: `as` assertion can be removed and comment above as well

File: packages/integrations/src/media-organizer/readarr/readarr-integration.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    start: Date,
+    end: Date,
+    includeUnmonitored = true,
+    includeAuthor = true,
+  ): Promise<CalendarEvent[]> {
+    const url = new URL(this.integration.url);
+    url.pathname = "/api/v1/calendar";
+    url.searchParams.append("start", start.toISOString());
+    url.searchParams.append("end", end.toISOString());
+    url.searchParams.append("unmonitored", includeUnmonitored ? "true" : "false");
```
Comment (Meierschlumpf) [inline packages/integrations/src/media-organizer/readarr/readarr-integration.ts:34]: Maybe instead use `includeUnmonitored.toString()`?

File: packages/integrations/src/media-organizer/readarr/readarr-integration.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    url.searchParams.append("includeAuthor", includeAuthor ? "true" : "false");
+    const response = await fetch(url, {
+      headers: {
+        "X-Api-Key": super.getSecretValue("apiKey"),
+      },
+    });
+    const readarrCalendarEvents = await z.array(lidarrCalendarEventSchema).parseAsync(await response.json());
```
Comment (Meierschlumpf) [inline packages/integrations/src/media-organizer/readarr/readarr-integration.ts:41]: Wrong named schema

File: packages/integrations/src/media-organizer/readarr/readarr-integration.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        date: readarrCalendarEvent.releaseDate,
+        mediaInformation: {
+          type: "audio",
+        },
+        links: this.getLinksForLidarrCalendarEvent(readarrCalendarEvent),
+      };
+    });
+  }
+
+  private getLinksForLidarrCalendarEvent = (event: z.infer<typeof lidarrCalendarEventSchema>) => {
```
Comment (Meierschlumpf) [inline packages/integrations/src/media-organizer/readarr/readarr-integration.ts:58]: Should be readarr

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

File: packages/integrations/src/media-organizer/readarr/readarr-integration.ts (hunk 2 above)
Comment (manuel-rw) [reply packages/integrations/src/media-organizer/readarr/readarr-integration.ts:34]: There is no such function

Comment (Meierschlumpf) [reply packages/integrations/src/media-organizer/readarr/readarr-integration.ts:34]: She is not autocompleted but works, you can try it. It even gives you a comment when hovering over it

**Review by Meierschlumpf — DISMISSED** [review]

**Review by Meierschlumpf — APPROVED** [review]

**Pull request merged**